src/irc/IrcServer.ts

```diff
@@ -502,6 +504,21 @@ export class IrcServer {
         return shouldSync;
     }
 
+    /**
+     * Does the server/channel require all Matrix users to be joined?
+     * @param channel The IRC channel.
+     * @returns True if the server requires all Matrix users to be joined.
+     */
+    public shouldRequireMatrixUserJoined(channel: string) {
+        let shouldSync = this.config.membershipLists.global.ircToMatrix.requireMatrixJoined;
+        this.config.membershipLists.channels.forEach((chan) => {
+            if (chan.channel === channel && typeof chan.ircToMatrix?.requireMatrixJoined === "boolean") {
```

Comment:
Isn't this a bit expensive? It iterates through all the channels even if the channel has already been found.

Reply:
Good catch.